libnmstate/nm/connection.py

```diff
@@ -182,241 +189,265 @@ def get_setting_duplicate(self, setting_name):
         return setting
 
     def _active_connection_callback(self, src_object, result, user_data):
-        cancellable = user_data
-        self._mainloop.drop_cancellable(cancellable)
+        if self._ctx.is_cancelled():
```

Comment:
We checked this originally only in the exception path, why is this now here? Is this consistent with the other callbacks?

Reply:
There is need to proceed when all is cancelled.

Yes. All callback functions do `is_cancelled` at the beginning.